### PR TITLE
Initial support to inline notices

### PR DESCRIPTION
**Pull request by claudiosanches**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This Pull Request enables initial support to inline notices, changing `wc_add_notice()` to accept an array of data (e.g. `['id' => 'billing_email']`), this data is displayed in the notice template (e.g `<ul class="woocommerce-error" role="alert"><li data-id="billing_email"> Billing Email is a required field.</li></ul>`), so in the future (or in 3rd party plugins or themes) will be possible to move those messages to each specific input in order to display inline notices.

### How to test the changes in this Pull Request:

1. Apply this PR
2. Try place an order missing to fill some fields or using an incorrect postcode (depends on the selected country to proper validate it, if you want to test it, you can test with US, UK, or Brazil)
3. Note that the notices display like before, but with the new HTML where includes `data-id`.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Added initial support for inline notices on checkout.

### CODE REVIEW

File: templates/notices/error.php [hunk 1]
```diff
@@ -10,26 +10,24 @@
...
 
 ?>
 <ul class="woocommerce-error" role="alert">
-	<?php foreach ( $messages as $message ) : ?>
-		<li>
-			<?php
-				echo wc_kses_notice( $message );
-			?>
+	<?php foreach ( $notices as $notice ) : ?>
+		<li<?php echo wc_get_notice_data_attr( $notice ); ?>>
```
Comment (rodrigoprimo) [inline templates/notices/error.php]: This is not a blocker, but I believe we need to add `wc_get_notice_data_attr()` to `WordPress.Security.EscapeOutput` in the `woocommerce-sniffs` repository?

**Review by rodrigoprimo — COMMENTED** [review]
Looks good to me. I left one non-blocking suggestion and I also added two more unit tests. Before merging this PR there are some conflicts that need to be resolved. @claudiosanches, could you please fix the conflicts?

File: templates/notices/error.php (hunk 1 above)
Comment (claudiosanches) [reply templates/notices/error.php]: Yes, that's my idea, to include on WooCommerce Sniffs after we merge it.

**Comment by claudiosanches** [comment]
@rodrigoprimo unit tests fixed.

**Review by rodrigoprimo — APPROVED** [review]

**Pull request merged**